### DISCUSSION TITLE
Copying files into images?

### DISCUSSION THREAD

**Discussion by jbehrends** [ZuluSCSI/ZuluSCSI-firmware#295] [Q&A] [answered]
Just picked up a ZuluSCSI RP2040 and had a simple question which I couldn't find asked here so figured I'd ask:

When I have the SD card in my Linux or Windows machine (outside of the ZuluSCSI device), is there any way to mount the .hda/.img files so I can easily copy files into the disk image?

**Comment by aperezbios** [comment]
@jbehrends without knowing the filesystem and partition map type contained within the image file, it's not possible to answer that question. Are you using your ZuluSCSI with a classic Mac, or something else?

**Reply by jbehrends** [reply]
That is a very fair point!  This is being used on an IBM PS/2 and formatted w/ MSDOS 6.22

**Comment by dranch** [comment, marked as answer]
I have a script that will do this mounting for simplistic use cases

http://www.trinityos.com/HAM/CentosDigitalModes/RPi/usr/local/sbin/mount-disk-image.sh
--
#!/bin/bash

# 11/18/22 - dranch mounting szbox disk image

#  If this doesn't work, run fdisk filename to see if an offset is required
#  fdisk -l file-image
#     Look at the START column and if it does NOT start at zero, record this number
#     Take that number and multiply by the sector size (say 512)
#
#  mount -o offset=<offset in bytes> nps-2010-emails.dd /media/manu/

#Optional Offset size - this is assuming 512 byte sectors
OFFSET=$((264*512))

if [ -z $1 ]; then
   echo -e "\nDisk image not provided.  Aborting\n"
   exit 1
fi

if [ $UID -ne 0 ]; then
   echo -e "\nYou must be root. Aborting\n"
   exit 1
fi 

if [ ! -d /media/image ]; then
   mkdir /media/image
fi

if [ -n $OFFSET ]; then
   echo -e "\nMounting $1 on /media/image with a offset of: $OFFSET"
   mount -o offset=$OFFSET $1 /media/image
   if [ $? -eq 0 ]; then
      echo -e "Success!"
     else
      echo -e "Failed"
   fi
  else
   echo -e "\nMounting $1 on /media/image without any offset"
   mount $1 /media/image
   if [ $? -eq 0 ]; then
      echo -e "Success!"
     else
      echo -e "Failed"
   fi
fi

echo " "
--

**Reply by jbehrends** [reply]
Ok! I didn't realize that the images it used were mountable with the standard linux mount command!  That's great news, I'll give this a try a littler late this evening!  Thanks for the help!

**Reply by aperezbios** [reply]
Well, _FAT/DOS MBR_ formatted images are. It depends on the partition table format and the filesystem contained within the image file, but yes, in the case of FAT-formatted images, Linux makes it trivial to mount these sorts of disk images. Under Windows, you can use something like WinImage, which is shareware.

**Reply by jbehrends** [reply]
This worked perfectly in Linux.  Thanks for the help!
